### ISSUE TITLE
Potential breaking change to Goodreads/Gatsby API

### ISSUE THREAD

**Issue by metamas** [jamesdanylik/gatsby-source-goodreads#1]
I could not get this to work at all. The only user error I thought I could be making was providing an incorrect `id` field to the `options` object for the plugin's `gatbsy-config` setup. But I don't think it was that. I see you haven't updated it in quite some time. Possibly a change to the Goodreads API that has it breaking?

I'm using `"gatsby": "^2.10.4"`... Maybe the plugin is incompatible there? 🤔 

_The error I kept getting._
![image](https://user-images.githubusercontent.com/2520253/61739171-c409ce80-ad40-11e9-8c05-046155fa353a.png)

**Comment by jamesdanylik** [comment]
Hey there!  I'm presently at work and won't _really_ be able to look into this for a few hours -- just wanted you to know I saw it, it's on my to-do list, and to give you my quick thoughts in case they help.

The current build status is available [here](https://travis-ci.org/jamesdanylik/www-jamesdanylik-com); as you can see, everything still looks good for my site (on the backend at least; I never did finish the front!).  A quick glance reveals I have my gatsby version set to "next" from back before the major version change, perhaps this has hidden issues from me?

As to settings, '**id**' should be a numerical goodreads _userid_ to scrape, while '**key**' must be the _key_ value provided by goodreads, **_NOT_** the _secret_ value!  For rough format, see below (scrambled data, won't actually work):

eg
```
...
    {
      resolve: "@jamesdanylik/gatsby-source-goodreads",
      options: {
        key: 'Oqfa_INVALID_c8TAwKFAP', // see https://www.goodreads.com/api/keys
        id: '88888888'
      }
    },
...
```

Hope this helps!  I'll check back in I can reproduce your issue later; please let me know if it gets resolved on your end before then.

Thanks!

**Comment by whatrocks** [comment]
I'm also seeing this issue, if that's helpful. My config matches your format -- and everything was working fine until yesterday. Wondering if Goodreads changed something on their side. 

```
error Plugin @jamesdanylik/gatsby-source-goodreads returned an error
Error: Invalid response from Goodreads: XML failed to parse
```

**Comment by jamesdanylik** [comment]
Hm, two people have seen this issue now, _something_ is probably up...

I still haven't been able to reproduce on my side, however.  I'll do a full upgrade of my site components over the weekend and see what issues that draws out.

**Comment by whatrocks** [comment]
Any luck on this @jamesdanylik ?

**Comment by jamesdanylik** [comment]
Still no luck yet -- but my time was a bit limited on this last weekend.  I'm gonna take another shot at it after work today.  If you get a chance, it might also be helpful to have you verify you encounter the same error across different users; I was thinking it might be possible one of the entries on your list may have some content in it that throws off the parser.  This would potentially explain the difficulty in reproduction.

I'll also try to do better at keeping progress posted here, sorry about that!

**Comment by jamesdanylik** [comment]
I've just finished updating my site to the latest versions of gatsby and all dependencies; unfortunately, I still can't reproduce the issue on my end.  Everything seems to be working well even with the new upgrades, so I can eliminate that as a cause of concern: see the [travis-ci test results](https://travis-ci.org/) which confirm.  Right now my theory is that there is something particular to the XML response you are getting which causes parsing to fail -- perhaps you read a book with an XML tag in the title or description in some funny way?  We'll probably need to capture that response to move forward on this issue.

**Comment by metamas** [comment]
@jamesdanylik I appreciate you investigating this. I'm currently just doing a weekly export of my Goodreads shelves, to meet my needs. In the future, I'll probably want to put time into figuring this out, but for now it's fine. When I do, I'll look closer at the responses and circle back to this issue (if it's still open).

**Comment by jamesdanylik** [comment]
@metamas absolutely, happy to look into it!  I'll leave this issue open so hopefully we can both loopback to it when you have a chance or if I can get it reproduced.

**Comment by whatrocks** [comment]
@jamesdanylik I am embarrassed to admit that I changed my scripts and forgot to `source` my env file in my new script. Sorry for the churn. It's working fine now, obviously. Really appreciate the quick turn around.

**Comment by jamesdanylik** [comment]
@whatrocks No worries at all; I have done exactly this more times than I care to admit.  Happy to hear it's working for you now... and thanks especially for reporting back about that!
